src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -169,6 +170,11 @@ CreateModifyPlan(Query *originalQuery, Query *query,
 {
 	if (InsertSelectQuery(originalQuery))
 	{
+		if (MultiTaskQueryLogLevel != MULTI_TASK_QUERY_INFO_OFF)
```

Comment:
An InsertSelectQuery can have only one task, so you also need to check task count on this. You can move this check into  CreateInsertSelectRouterPlan() and also check task count.